pkg/controllers/updaterun/execution.go

```diff
@@ -97,6 +93,10 @@ func (r *Reconciler) execute(
 			markUpdateRunWaiting(updateRun, fmt.Sprintf(condition.UpdateRunWaitingMessageFmt, "before-stage", updatingStageStatus.StageName))
 			return false, stageUpdatingWaitTime, nil
 		}
+		maxConcurrency, err := calculateMaxConcurrencyValue(updateRunStatus, updatingStageIndex)
```

Comment:
why is maxConcurrency calculation included in this PR?

Reply:
This change is from a different PR, it's just reordering code to ensure we calculateMaxConcurrencyValue before it's needed instead of calling it in a place where it doesn't fit into the flow